File: src/main/java/com/wishai/xzrtw/config/XzrtConfig.java
package com.wishai.xzrtw.config;

import com.wishai.xzrtw.component.CustomMultipartResolver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
import org.springframework.web.filter.CorsFilter;

import java.util.Arrays;


@Configuration
@EnableConfigurationProperties(XzrtProperties.class)
public class XzrtConfig {

    private final XzrtProperties config;

    @Autowired
    public XzrtConfig(XzrtProperties config) {
        this.config = config;
    }

    @Bean
    public CustomMultipartResolver multipartResolver() {
        CustomMultipartResolver multipartResolver = new CustomMultipartResolver();
        multipartResolver.setMaxUploadSize(500000000);
        return multipartResolver;
    }

    @Bean
    public CorsFilter corsFilter() {
        CorsConfiguration config = new CorsConfiguration();
        config.setAllowCredentials(true);
        config.setAllowedOrigins(Arrays.asList(this.config.getForwardRootUrl()));
        config.addAllowedHeader("*");
        config.addAllowedMethod("GET");
        config.addAllowedMethod("POST");

        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        source.registerCorsConfiguration("/**", config);
        return new CorsFilter(source);
    }
}

File: src/main/java/com/wishai/xzrtw/config/XzrtConfig.kt
package com.wishai.xzrtw.config

import com.wishai.xzrtw.component.CustomMultipartResolver
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.filter.CorsFilter
import java.util.*

@Configuration
@EnableConfigurationProperties(XzrtProperties::class)
class XzrtConfig @Autowired
constructor(private val config: XzrtProperties) {

    @Bean
    fun multipartResolver(): CustomMultipartResolver {
        val multipartResolver = CustomMultipartResolver()
        multipartResolver.setMaxUploadSize(500000000)
        return multipartResolver
    }

    @Bean
    fun corsFilter(): CorsFilter {
        // initialize config
        val config = CorsConfiguration()
        config.allowCredentials = true

        // add allowed origins and sources
        val origins = this.config.forwardRootUrl!!.split("^")
        config.allowedOrigins = origins
        config.addAllowedHeader("*")
        config.addAllowedMethod("GET")
        config.addAllowedMethod("POST")

        // apply config to sources
        val source = UrlBasedCorsConfigurationSource()
        source.registerCorsConfiguration("/**", config)
        return CorsFilter(source)
    }
}
